massa-consensus-worker/src/worker/init.rs

```diff
@@ -200,6 +200,13 @@ impl ConsensusWorker {
                     final_blocks.iter().map(|(b, _)| b.block_id).collect();
                 write_shared_state.best_parents = latest_final_blocks_periods.clone();
                 write_shared_state.latest_final_blocks_periods = latest_final_blocks_periods;
+                for (b, _) in &final_blocks {
```

Comment:
no need to bother with this, we discard incoming blocks that are at or before the genesis period

Reply:
Done.